### PR TITLE
Analyzing effect of dimensionality reduction on accuracy of different classifiers on different types of datasets

### PR DESCRIPTION
**Pull request by parimal173**
It will be a document showing the Effect of Dimensionality Reduction on the accuracy of different classifiers.
The document will have simulations On High dimensional dataset of different shapes:
Each dataset is synthesized from sklearn established Datasets. Each dataset has 100 dimensions with only 2 dimensions of data and rest are noise dimensions.

Questions to answer:

Analyzing how dimensionality reduction affects in classification for different classifiers.
Analyzing how classifiers perform with keeping different number of principle components of the main high dimensional dataset.
Pipeline to be followed:

Defining a dataset with sklearn established synthetic datasets with High dimensions.
Performing classification on data and measuring accuracy for quantification of the process.
performing the Dimensionality reduction technique keeping varying numbers of reduced dimensions.
Checking the performance of classification again after reducing dimension after each iteration.
The output of the PR would be a figure showing different datasets,a plot showing varying accuracies over reduced dimensions.

Experiments to follow:
https://github.com/parimal173/scikit-learn/blob/parimal173-patch-1/Final_PR_9figures_NonStandardNoise.ipynb

### CODE REVIEW

**Comment by bdpedigo** [comment]
@parimal173

> - in cell run [4] title the datasets and remove axes ticks and ticklabels
- did not address, remove the ticks and tick labels!
> - the main block of your code, cell run [10] is extremely hard code to read because of how many lists and things you have going on, and It is impossible to read stuff like dsvm1 += dim_data1[(198 * (i+1)):(297 * (i+1))] and know what is going on. Never hard code numbers like this. You should be able to do all of this a lot more succinctly with dataframes, or, look at how other people on your team have structured similar simulations.
- this main block is still impossible to read, you shouldn't need this many lists of data. I know this seems nitpicky but because of this i have a hard time following the code. should never need if statements like this.
- you don't need to do PCA 100 times, just do it once and take different numbers of components
- fix the last plot - most text way too small. Your # of dimensions is overlapping with ticks. I'd also make that say # of principal components kept or something more clear like that
- still don't know how much i trust the results but that is something to revisit more next semester at this point.

**Comment by bdpedigo** [comment]
@parimal173 
- remove all unnecessary files (i think there is only one that you want to PR?)
- give that file a descriptive name 
then i'll merge

**Comment by parimal173** [comment]
> @parimal173
> 
> * remove all unnecessary files (i think there is only one that you want to PR?)
> * give that file a descriptive name
>   then i'll merge

please tell me if the name is descriptive enough.

**Review by bdpedigo — APPROVED** [review]
